tools/parsec/parsec.xml

```diff
@@ -0,0 +1,151 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<tool id="batch_cohort_correction" name="Batch-Cohort Correction" version="1.1" profile="16.01">
+    <description>Corrects for batch and cohort effects on ion intensities using a mixed-model approach</description>
+
+    <command>
+        <![CDATA[
+            Rscript '$__tool_directory__/parsec.R' -d '$dataMatrix' -s '$sampleMetadata' -v '$variableMetadata' -o '$output'
+        ]]>
+    </command>
+
+    <inputs>
+        <param name="dataMatrix" type="data" format="tabular" label="Data Matrix (samples × variables)" />
+        <param name="sampleMetadata" type="data" format="tabular" label="Sample Metadata" />
+        <param name="variableMetadata" type="data" format="tabular" label="Variable Metadata" />
+    </inputs>
+
+    <outputs>
+        <data name="output" format="csv" label="Corrected Intensities Table" />
+    </outputs>
+
+    <tests>
+        <!-- Test with valid example files -->
+        <test>
+            <param name="dataMatrix" value="Dataprocessing_dataMatrix.txt" />
+            <param name="sampleMetadata" value="Dataprocessing_sampleMetadata.txt" />
+            <param name="variableMetadata" value="Dataprocessing_variableMetadata.txt" />
+            <output name="output" file="corrected_output_test.csv" />
+        </test>
+
+        <!-- Test: Missing injection order -->
+        <test expect_failure="true">
+            <param name="dataMatrix" value="missing_injectionOrder_matrix.txt" />
+            <param name="sampleMetadata" value="Dataprocessing_sampleMetadata.txt" />
+            <param name="variableMetadata" value="Dataprocessing_variableMetadata.txt" />
+        </test>
+
+        <!-- Test: Empty file -->
+        <test expect_failure="true">
+            <param name="dataMatrix" value="empty_file.txt" />
+            <param name="sampleMetadata" value="Dataprocessing_sampleMetadata.txt" />
+            <param name="variableMetadata" value="Dataprocessing_variableMetadata.txt" />
+        </test>
+    </tests>
+
+    <requirements>
+        <requirement type="package" version="4.4.2">r-base</requirement>
+        <requirement type="package" version="1.7.5">r-optparse</requirement>
+        <requirement type="package" version="1.1.4">r-dplyr</requirement>
+        <requirement type="package" version="1.1.36">r-lme4</requirement>
+    </requirements>
+
+    <help><![CDATA[
```

Comment:
```suggestion
    <help format="markdown"><![CDATA[
```

That should work in newer Galaxy version. If not, you need to use restructured text.